### PR TITLE
Bes keys fix

### PR DESCRIPTION
**Pull request by jgallagher59701**
TheBESKeys now uses a unique_ptr and passes tests.
    
Some other code had to be changed because the singleton
was 'messed with' by some of the BES framework destructors
and some of the unit-tests. I added a reload_keys() method
to enable unit-tests to switch key files and/or reload the
original key file, effectively resetting changes they might
have made during the tests.

### CODE REVIEW

**Comment by kyang2014** [comment]
Does this fix the memory leak caused by the BES key operations?

File: dispatch/unit-tests/old/keysT.cc [hunk 1]
```diff
@@ -279,9 +243,6 @@ class keysT : public TestFixture {
         if (debug) cout << __func__ << END << endl;
     }
 
-    /**
```
Comment (ndp-opendap) [inline dispatch/unit-tests/old/keysT.cc:282]: What's the deal with the method comment headers? It helps me find each method... :(

**Review by ndp-opendap — APPROVED** [review]

File: dispatch/unit-tests/old/keysT.cc (hunk 1 above)
Comment (jgallagher59701) [reply dispatch/unit-tests/old/keysT.cc:282]: Oh, sorry. I thought those were just auto generated and not filled in. Should I add them back in?

Comment (jgallagher59701) [reply dispatch/unit-tests/old/keysT.cc:282]: Wait, that's the keysT.cc code - it's now in 'old' (aka unused).

**Pull request merged**